politeiad/backendv2/tstorebe/plugins/pi/hooks.go

```diff
@@ -390,11 +395,246 @@ func (p *piPlugin) voteSummary(token []byte) (*ticketvote.SummaryReply, error) {
 	return &sr, nil
 }
 
+// comments requests all comments on a record from the comments plugin.
+func (p *piPlugin) comments(token []byte) (*comments.GetAllReply, error) {
+	reply, err := p.backend.PluginRead(token, comments.PluginID,
+		comments.CmdGetAll, "")
+	if err != nil {
+		return nil, err
+	}
+	var gar comments.GetAllReply
+	err = json.Unmarshal([]byte(reply), &gar)
+	if err != nil {
+		return nil, err
+	}
+	return &gar, nil
+}
+
+// isAncestorOf returns whether the given parentID is an ancestor of the given
+// childID comment, by traveling the comment tree defined by
+// the ParentID of the child comment.
+func (p *piPlugin) isAncestorOf(ancestorID, childID uint32, cs []comments.Comment) bool {
+	if ancestorID == childID {
+		return true
```

Comment:
What's the reasoning behind this?  To me this introduces unexpected behavior.  I would not normally consider a comment to be included as an ancestor of itself.

Reply:
right, i passed the comment.parentID to the func, instead i pass the comment id of the new comment
and then compare: `ancestorID == current.ParentID`